itensor/itensor_interface.h

```diff
@@ -117,12 +121,14 @@ class ITensorT
     template<typename IV, typename... VArgs>
     auto
     set(IV const& iv1, VArgs&&... ivs)
-        -> stdx::if_compiles_return<void,decltype(iv1.index),decltype(iv1.val)>;
```

Comment:
Is this ok to remove? I'm not sure what it is doing.

Reply:
Hi Matt,
Let's leave that unless it's causing an issue. Rule #1 of editing a library is that if you don't know what a line of code is doing, it's probably some bug fix for a rare weird case that happened once long ago. (Applies also to code you yourself wrote then forgot why.)

What that line looks to be doing is making sure that "IV" is either the IndexVal or IQIndexVal type, since those types have a ".index" and ".val" field.

Reply:
Definitely a good rule. I changed it back, I thought I had to change it to get it to dispatch properly on the different versions of .set but it is working properly with it there.